pkg/async/notifications/factory.go

```diff
@@ -148,3 +148,57 @@ func NewNotificationsPublisher(config runtimeInterfaces.NotificationsConfig, sco
 		return implementations.NewNoopPublish()
 	}
 }
+
+func NewEventsPublisher(config runtimeInterfaces.NotificationsConfig, scope promutils.Scope) interfaces.Publisher {
+	if config.ExternalEvent.EventPublisherConfig.TopicName == "" {
```

Comment:
can we explicitly just use the `Type` of the Notificationsconfig. and if it's not no-op, error when Topic Name is empty?